### ISSUE TITLE
epub3 validation error caused by sphinx.ext.inheritance_diagram

### ISSUE THREAD

**Issue by juh2** [sphinx-doc/sphinx#4501]
Subject: I am validating a project with an inheritance-diagram

### Problem
- When validating the epub with epubcheck 4.0.2 I get a validation error.

#### Procedure to reproduce the problem

Simply build a diagram eg.

```
.. inheritance-diagram:: sphinx.builders.html

```

#### Error logs / results
```
Error while parsing file 'element "area" missing required attribute "href"'.
```

#### Expected results

validation without errors

### Environment info
- OS: <Unix/Linux/Mac/Win/other with version>
- Python version: 3.5.3
- Sphinx version: 1.6.6
- <Extra tools e.g.: Browser, tex or something else>

**Comment by tk0miya** [comment]
Hmm? HTML5 spec says the href attribute is optional.

>If the area element has an href attribute, then the area element represents a hyperlink. In this case, the alt attribute must be present. It specifies the text of the hyperlink. Its value must be text that informs the user about the destination of the link.
>
>If the area element has no href attribute, then the area represented by the element cannot be selected, and the alt attribute must be omitted.
>https://www.w3.org/TR/html52/semantics-embedded-content.html#the-area-element

I have to know why epubcheck warns this.

**Comment by shibukawa** [comment]
I am checking the issue. inheritance-diagram extension calls graphviz extension inside it. The HTML fragments that includes <area> tags is generated via Graphviz and imported at the following location. To fix the issue, we have to parse the HTML and modify the content.

https://github.com/sphinx-doc/sphinx/blob/stable/sphinx/ext/graphviz.py#L257

**Comment by tk0miya** [comment]
Fixed by #4515 
Thank you for reporting!
